### ISSUE TITLE
scalar bounds get wrong size

### ISSUE THREAD

**Issue by sfleuren** [coin-or/CyLP#68]
In the newest version (0.9.0b1) I encountered the following error. Consider the following problem which is part of the unittests 'test_isInt' from https://github.com/coin-or/CyLP/blob/master/cylp/tests/test_MIP.py:

```
x = model.addVariable('x', 3, isInt=True)
y = model.addVariable('y', 2)

A = np.matrix([[1., 2., 0],[1., 0, 1.]])
B = np.matrix([[1., 0, 0], [0, 0, 1.]])
D = np.matrix([[1., 2.],[0, 1]])
a = CyLPArray([5, 2.5])
b = CyLPArray([4.2, 3])
x_u= CyLPArray([2., 3.5])

model.addConstraint(A*x <= a)
model.addConstraint(2 <= B * x + D * y <= b)
model.addConstraint(y >= 0)
model.addConstraint(1.1 <= x[1:3] <= x_u)

c = CyLPArray([1., -2., 3.])
model.objective = c * x + 2 * y.sum()
```
I get the error:
ValueError: shape mismatch: value array of shape (3,) could not be broadcast to indexing result of shape (2,)

Digging in the code I think the following happens:
For the left-hand-side of the constraint 'model.addConstraint(1.1 <= x[1:3] <= x_u)' a vector with the same size as the variable x is created (which is 3), while it should have the same size as x[1:3] (which is 2).

Regards, Stijn

**Comment by tkralphs** [comment]
Thanks for the report. I hadn't had a chance yet to dig into these tests with the new beta version. If you can submit a PR, that would save me some time. Otherwise, I'm sure I can pretty easily figure out the fix if I just step through this example. Strange that it happens, since this part of the code shouldn't have changed and must have worked at some point.

**Comment by sfleuren** [comment]
I believe I am not familiar enough with the code to be able to make this fix. 
Only thing I noticed is that the example fails on line 523 of cylp\py\modeling\CyLPModel.py
`var.parent.lower[var.indices] = bound`
At this point var.indices equals [1,2] (which are the indices of x that are specified). And bound=[1.1,1.1,1.1], which is the lower bound of this constraint (with one element too much).

**Comment by tkralphs** [comment]
This should be fixed by af1abbb773. There was a comment above the section I changed that said
```
# FIXME: check this: I suppose never runs
```
although the code was actually reached in this case. I'm a bit mystified, since the original developer did write these tests and would have run into this bug long back. Also, the fix was rather simple, which makes me worry :). But the 5 tests in that file all seem to pass now.
